lib/oli/automation_setup.ex

```diff
@@ -0,0 +1,354 @@
+                 :communities,
+                 :publications
+               ]
+           )
+           |> has_project(),
+         # Only tear down projects where duplicates are not allowed
+         {:ok} <- no_duplicates(project),
+         # Only tear down projects with no authors
+         {:ok} <-
+           has_no_authors(project.authors) do
+      #
+      # Delete all the publications for this project, this will also get the publication_resources
+      Enum.each(project.publications, &Oli.Publishing.delete_publication/1)
+
+      # Delete all the associations between this project and resources
+      Repo.delete_all(from pr in ProjectResource, where: pr.project_id == ^project.id, select: pr)
+
+      # Delete the resources
+      for resource <- project.resources do
+        Repo.delete_all(from r in Oli.Resources.Revision, where: r.resource_id == ^resource.id)
+        Repo.delete(resource)
+      end
+
+      {:ok, _} = Repo.delete(project)
+
+      %{success: true}
+    else
+      {:error, message} -> %{success: false, message: message}
+      _ -> %{success: false, message: "Unknown Reason"}
+    end
+  end
+
+  def teardown_section(nil) do
+    %{success: false, message: "No section slug provided"}
+  end
+
+  def teardown_section(slug) do
+    with {:ok, section} <-
+           Oli.Delivery.Sections.get_section_by(slug: slug) |> can_delete_section(),
+         {:ok, _} <- Oli.Delivery.Sections.delete_section(section) do
+      %{success: true}
+    else
+      {:error, message} -> %{success: false, message: message}
+      _ -> %{success: false, message: "Unknown Reason"}
+    end
+  end
+
+  def teardown_educator(email, password) do
+    teardown_user(email, password, :user, "Test Educator")
+  end
+
+  def teardown_learner(email, password) do
+    teardown_user(email, password, :user, "Test Learner")
+  end
+
+  def teardown_author(email, password) do
+    teardown_user(email, password, :author, "Test Author")
+  end
+
+  defp teardown_user(nil, _, _, _),
+    do: %{
+      success: false,
+      message: "User not specififed"
+    }
+
+  defp teardown_user(_, nil, _, _),
+    do: %{
+      success: false,
+      message: "Password not specified"
+    }
+
+  defp teardown_user(email, password, user_type, expected_name) do
+    case(validate_user(email, password, user_type, expected_name)) do
+      {:error, message} ->
+        %{
+          success: false,
+          message: message
+        }
+
+      {:ok, user} ->
+        Logger.info("Deleting test user #{email}")
+        Oli.Repo.delete(user)
+        %{success: true}
+    end
+  end
+
+  defp validate_user(email, password, user_type, expected_name) do
+    config = OliWeb.Pow.PowHelpers.get_pow_config(user_type)
+
+    case Pow.Operations.get_by([{:name, expected_name}, {:email, email}], config) do
+      nil ->
+        {:error, "User not found"}
+
+      user ->
+        case user.__struct__.verify_password(user, password) do
```

Comment:
I'm not entirely sure this is the best way to validate user credentials. Searching through POW api docs, I found a lot of ways of authenticating a user based on their session or through a plug validating form values, but I came up short on finding a generic email/pass check.